examples/static-contact-dynamics.py

```diff
@@ -0,0 +1,166 @@
+import pinocchio as pin
+
+from os.path import dirname, join, abspath
+np.set_printoptions(linewidth=np.inf)
+
+# ----- PROBLEM STATEMENT ------
+#
+# We want to find the contact forces and torques required to stand still at a configuration 'q0'.
+# We assume 3D contacts at each of the feet
+#
+# The dynamic equation would look like:
+#
+# M*q_ddot + g(q) + C(q, q_dot) = tau + J^T*lambda --> (for the static case) --> g(q) = tau + Jc^T*lambda (1)
+
+# ----- SOLVING STRATEGY ------
+
+# Split the equation between the base link (_bl) joint and the rest of the joints (_j). That is,
+#
+#  | g_bl |   |  0  |   | Jc__feet_bl.T |   | l1 |
+#  | g_j  | = | tau | + | Jc__feet_j.T  | * | l2 |    (2)
+#                                           | l3 |
+#                                           | l4 |
+
+# First, find the contact forces l1, l2, l3, l4 (these are 3 dimensional) by solving for the first 6 rows of (2).
+# That is,
+#
+# g_bl   = Jc__feet_bl.T * | l1 |
+#                          | l2 |    (3)
+#                          | l3 |
+#                          | l4 |
+#
+# Thus we find the contact froces by computing the jacobian pseudoinverse,
+#
+# | l1 | = pinv(Jc__feet_bl.T) * g_bl  (4)
+# | l2 |
+# | l3 |
+# | l4 |
+#
+# Now, we can find the necessary torques using the bottom rows in (2). That is,
+#
+#                             | l1 |
+#  tau = g_j - Jc__feet_j.T * | l2 |    (5)
+#                             | l3 |
+#                             | l4 |
+
+# ----- SOLUTION ------
+
+# 0. DATA
+pinocchio_model_dir = join(dirname(dirname(str(abspath(__file__)))), "models")
+
+model_path = join(pinocchio_model_dir, "example-robot-data/robots")
+mesh_dir = pinocchio_model_dir
+urdf_filename = "solo12.urdf"
+urdf_model_path = join(join(model_path, "solo_description/robots"),
+                       urdf_filename)
+
+model, collision_model, visual_model = pin.buildModelsFromUrdf(
+    urdf_model_path, mesh_dir, pin.JointModelFreeFlyer())
+data = model.createData()
+
+q0 = np.array([
+    0., 0., 0.235, 0., 0., 0., 1., 0., 0.8, -1.6, 0., -0.8, 1.6, 0., 0.8, -1.6,
+    0., -0.8, 1.6
+])
+v0 = np.zeros(model.nv)
+a0 = np.zeros(model.nv)
+
+# 1. GRAVITY TERM
+
+# We compute the gravity terms by using the ID at desired configuration q0, with velocity and acceleration being 0. I.e., ID with a = v = 0
+g_grav = pin.rnea(model, data, q0, v0, a0)
+
+g_bl = g_grav[:6]
+g_j = g_grav[6:]
+
+# 2. FIND CONTACTS
+
+# First, we set the frame for our contacts. We assume the contacts are placed at the following 4 frames and they are 3D
+feet_names = ['FL_FOOT', 'FR_FOOT', 'HL_FOOT', 'HR_FOOT']
+feet_ids = [model.getFrameId(n) for n in feet_names]
+bl_id = model.getFrameId('base_link')
+ncontact = len(feet_names)
+
+# Now, we need to find the contact Jacobians appearing in (1).
+# These are the Jacobians that relate the joint velocity  to the velocity of each feet
+Js__feet_q = [
+    np.copy(pin.computeFrameJacobian(model, data, q0, id, pin.LOCAL))
```

Comment:
Compute the forces in "LOCAL_WORLD_ALIGNED" coordinates as this is what we typically want.
You can use the LOCAL_WORLD_ALIGNED contact Jacobian or multiply by the rotation matrix the linear force.

Reply:
Working with both is fine I would say. No need to change here.